solver/solve.go

```diff
@@ -5,6 +5,7 @@ import (
 	"encoding/json"
 
 	"github.com/docker/buildx/util/progress"
+	"github.com/docker/docker/api/types/container"
```

Comment:
No particular opinion either way, but if you feel it's worth avoiding the dependency on `docker/docker` (which is a fairly large repo IIRC), you could use something like:

```go
type ContainerConfig struct {
        Cmd      []string
        Labels   map[string]string
}
```